Extensions/Flows/Handlers/AdditionalStylingPartHandler.cs

```diff
@@ -18,7 +19,7 @@ public override Task ActivatedAsync(ActivatedContentContext context)
         {
             if (!context.ContentItem.Has<AdditionalStylingPart>() &&
                 _contentDefinitionManager.GetTypeDefinition(context.ContentItem.ContentType)
-                    .GetSettings<ContentTypeSettings>().Stereotype == "Widget")
+                    .GetSettings<ContentTypeSettings>().Stereotype?.EqualsOrdinal("Widget") == true)

```

Comment:
This is the thing I consider reversing the whole `EqualsOrdinal()` thing. The equals sign doesn't bother about `null` but this does. Not sure what to do. Maybe a different `EqualsOrdinal()`? I know we had a null check there previously but pulled it out for the sake of consistency with the framework.

Reply:
I'm actually starting to lean the other direction, that we should always assume that the `this` parameter in an extension method is never null (this will be easier in .Net 6 where nullable is on by default). Member methods expect their object to exist, so code becomes pretty irregular when you have custom logic for `null` valued `this` parameter. It means using the null-conditional operator skips that extra logic **so you actually have to pay attention for when to use it and discriminate extension methods from regular methods** which arguably defeats the purpose.

If you aren't happy with the change in this line, it would be better to create a static helper class that adds custom logic where we aren't bound to following the conventions in existing framework methods.

btw stuff like this really makes me wish that C# had custom infix operators like in R.

Reply:
As mentioned below, `.EqualsOrdinal()` actually does not do anything else than `==`.

I like the idea of keeping the null check out of the extension method. With the null-conditional operator at hand, this is an unintrusive decision when it comes to code-style, but with the plus of consistency with instance methods.

Reply:
So this is now not a question anymore, see https://github.com/Lombiq/Helpful-Extensions/pull/47#discussion_r775775908.

Reply:
Is it not applicable to the EqualsOrdnialIgnoreCase method?

Reply:
On the null question I agree with both of you, i.e. keep the current methods as they are and not do null checks. They should throw as they do today.